sound/soc/sof/pm.c

```diff
@@ -306,7 +306,7 @@ int snd_sof_prepare(struct device *dev)
 {
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
 
-#if defined(CONFIG_ACPI)
+#if !defined(CONFIG_SOF_OF) && defined(CONFIG_ACPI)
```

Comment:
!IS_ENABLED(CONFIG_SOF_OF) ?

Reply:
I don't think this works anyways, if someone enabled ACPI in your build it breaks your power management. too fragile.

I think we should add a boolean in the platform descriptors that defines if pm is managed by ACPI.

Reply:
@plbossart isn't the logic here exactly the opposite? As far as I understand this PR enables RT-PM on i:MX8, i.e. with `CONFIG_SOF_OF` enabled, right? Then according to this `#if` it doesn't matter whether `CONFIG_ACPI` is enabled - it anyway will take the `#else` branch

Reply:
@lyakh @plbossart my intention is as @lyakh says. If CONFIG_SOF_OF is enabled then snd_sof_prepare will always select S3 by default as system_suspend_target. If CONFIG_SOF_OF is not enabled the behavior is unchanged.

Reply:
@plbossart this can be an alternative as you suggested -> https://github.com/thesofproject/linux/pull/2441